R/timeseries_plot.R

```diff
@@ -0,0 +1,98 @@
+#' Title
+#'
+#' @param data 
+#' @param time 
+#' @param elements 
+#' @param station 
+#' @param facets 
+#' @param add_points 
+#' @param add_line_of_best_fit 
+#' @param se 
+#' @param add_path 
+#' @param add_step 
+#' @param na.rm 
+#' @param show.legend 
+#'
+#' @return
+#' @export
+#'
+#' @examples
+timeseries_plot <- function(data, time, elements, station = NULL, facets = c("none", "elements", "stations", "both"),
+                            add_points = FALSE, add_line_of_best_fit = FALSE,
+                            se = TRUE, add_path = FALSE, add_step = FALSE,
+                            na.rm = FALSE, show.legend = NA){
+  
+  checkmate::assert_data_frame(data)
+  # time can be a date, factor, or character.
+  # call date_time for consistency with summary plot?
+  checkmate::assert_character(elements)
+  checkmate::assert_character(station, null.ok = TRUE)
+  facets <- match.arg(facets)
+  checkmate::assert_logical(add_points)
+  checkmate::assert_logical(add_line_of_best_fit)
+  checkmate::assert_logical(se)
+  checkmate::assert_logical(add_path)
+  checkmate::assert_logical(add_step)
+  checkmate::assert_logical(show.legend)
+  
+  data_longer <- data %>% tidyr::pivot_longer(cols = c({{ elements }}), names_to = "elements")
+  if (facets == "elements"){
+    if (is.null(.data[[station]])){
+      base_plot <- ggplot2::ggplot(data_longer, mapping = ggplot2::aes(x = .data[[time]], y = value)) +
+        ggplot2::facet_grid(cols = vars(elements))
+    } else {
+      base_plot <- ggplot2::ggplot(data_longer, mapping = ggplot2::aes(x = .data[[time]], y = value, colour = .data[[station]])) +
+        ggplot2::facet_grid(cols = vars(elements))
+    }
+  } else if (facets == "stations"){
+    if (length(elements) == 1){
+      base_plot <- ggplot2::ggplot(data, mapping = ggplot2::aes(x = .data[[time]], y = .data[[elements]]))
+    } else {
+      base_plot <- ggplot2::ggplot(data_longer, mapping = ggplot2::aes(x = .data[[time]], y = value, colour = elements))
```

Comment:
```suggestion
      base_plot <- ggplot2::ggplot(data_longer, mapping = ggplot2::aes(x = .data[[time]], y = value, colour = .dataelements))
```

Reply:
Here, `elements` is a new variable which is created in `data_longer`. It is all the elements binded into one variable so that we can separate by colour. So I don't think your suggestion is what we want.

Reply:
I will do `.data$elements` for this reason

Reply:
Ah I see, yes you're right. But to avoid confusion I think it's best not to create a variable with the same name as an argument to the function. Could this be renamed?